service/history/ndc/workflow_resetter_test.go

```diff
@@ -730,7 +732,10 @@ func (s *workflowResetterSuite) TestReapplyContinueAsNewWorkflowEvents_WithConti
 	mutableState := workflow.NewMockMutableState(s.controller)
 	mutableState.EXPECT().VisitUpdates(gomock.Any()).Return()
 	mutableState.EXPECT().GetCurrentVersion().Return(int64(0))
-	updateRegistry := update.NewRegistry(mutableState)
+	mutableState.EXPECT().GetWorkflowKey().Return(definition.WorkflowKey{RunID: "random-run-id"})
```

Comment:
not sure if there's a specific run id I should put here to make this test actually cover the scenario we're fixing. it's a bit hard to know with all the mocks

Reply:
I actually don't know what this test is doing. It mixes reset run, base run and new run. There should be 3 mutable state defined here but I only see two.

- Current run should be newRun and the mock here should use `newRunID`. 
- The resetMutableState should not use `newRunID` on L723. 
- It should be `resetMutableState` on L747
- we need to lock the mutable state for the newRunID